lib/codecs.js

```diff
@@ -0,0 +1,23 @@
+var iconv = require('iconv-lite');
+
+var cache = {};
+
+function getCodec(encoding) {
+  var codec = cache[encoding];
+  if (!!codec || !encoding || cache.hasOwnProperty(encoding)) {
+    return codec;
+  }
+
+  try {
```

Comment:
I figured we would cache the failed cases too -- that way we won't have to go into a try/catch repeatedly. The main thing is this function returns undefined when no codec is found, which is then used by the callers to emit an actual error.

Perhaps this new file needs a few tests as well, I might find some time today.